README.md

```diff
@@ -1,38 +1,155 @@
+  [![Check Workflow Status](https://github.com/keep-starknet-strange/snos/actions/workflows/check.yml/badge.svg)](https://github.com/keep-starknet-strange/snos/actions/workflows/check.yml)
+[![license](https://img.shields.io/github/license/keep-starknet-strange/snos)](/LICENSE)
+[![pr-welcome]](#-contributing)
+
+[pr-welcome]: https://img.shields.io/static/v1?color=blue&label=PRs&style=flat&message=welcome
 
-</h1>
 </div>
 
-Rust Library for running the [Starknet OS](https://hackmd.io/@pragma/ByP-iux1T) via the [Cairo VM](https://github.com/lambdaclass/cairo-vm).
+## Table of Contents
+- [Table of Contents](#table-of-contents)
+- [📖 About](#-about)
+- [🛠️ Getting Started](#️-getting-started)
+  - [Prerequisites](#prerequisites)
+  - [Installation](#installation)
+- [🧪 Running Tests](#-running-tests)
+  - [Run Tests](#run-tests)
+  - [Reset Tests](#reset-tests)
+- [🚀 Usage](#-usage)
+  - [Adding SNOS as a Dependency](#adding-snos-as-a-dependency)
+  - [Using the **prove_block** Binary](#using-the-prove_block-binary)
+- [🤝 Related Projects](#-related-projects)
+- [📚 Documentation](#-documentation)
+- [📜 License](#-license)
+
+
+## 📖 About
+
+[Starknet OS](https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/core/os/os.cairo) is a [Cairo](https://www.cairo-lang.org/) program designed to prove the integrity of state transitions between blocks on Starknet.
+
+By re-executing transactions from a block and verifying consistency, it produces a [PIE](https://github.com/starkware-libs/cairo-lang/blob/a86e92bfde9c171c0856d7b46580c66e004922f3/src/starkware/cairo/lang/vm/cairo_pie.py#L219-L225) (Program Independent Execution) result. This PIE can be used to generate a Stark proof of integrity, which, if accepted by Starknet L1 verifiers, confirms block validity and updates the Starknet state root in the [StarknetCore contract](https://etherscan.io/address/0xc662c410c0ecf747543f5ba90660f6abebd9c8c4#code).
+
+## 🛠️ Getting Started
 
-## Test Setup
+### Prerequisites
 
-**Cairo Env**
-(see: https://docs.cairo-lang.org/0.12.0/quickstart.html)
+Ensure you have the following dependencies installed:
+- [Rust 1.76.0 or newer](https://www.rust-lang.org/tools/install)
+- Cargo (included with Rust)
```

Comment:
Redundant

Reply:
It's installed as standard - the link you provide will install rustup and also cargo.